configure_cmd.go

```diff
@@ -0,0 +1,133 @@
+// provided URL is URL of Satellite server.
+//
+// We would like to use different approach in the future. We would like to use
+// some API endpoints not restricted by username & password for getting CA certs
+// and rendered rhsm.conf, because it would be more secure, but it is not possible ATM
+func satelliteAction(ctx *cli.Context) error {
+	var configureSatelliteResult ConfigureSatelliteResult
+	configureSatelliteResult.format = ctx.String("format")
+	satelliteUrlStr := ctx.String("url")
+
+	uid := os.Getuid()
+	if uid != 0 {
+		errMsg := "non-root user cannot connect system"
+		exitCode := 1
+		return cli.Exit(fmt.Errorf("error: %s", errMsg), exitCode)
+	}
+
+	hostname, err := os.Hostname()
+	if uiSettings.isMachineReadable {
+		configureSatelliteResult.Hostname = hostname
+	}
+	if err != nil {
+		exitCode := 1
+		if uiSettings.isMachineReadable {
+			configureSatelliteResult.HostnameError = err.Error()
+			return cli.Exit(configureSatelliteResult, exitCode)
+		} else {
+			return cli.Exit(err, exitCode)
+		}
+	}
+
+	satelliteUrl, err := normalizeSatelliteScriptUrl(satelliteUrlStr)
+	if err != nil {
+		return fmt.Errorf("could not parse satellite url: %w", err)
+	} else {
+		configureSatelliteResult.SatelliteServerHostname = satelliteUrl.Hostname()
+		configureSatelliteResult.SatelliteServerScriptUrl = satelliteUrl.String()
+	}
+
+	var s *spinner.Spinner = nil
+	if uiSettings.isRich {
+		s = spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+		s.Suffix = fmt.Sprintf(" Configuring '%v' to use Satellite %v", hostname, satelliteUrl.Host)
+		s.Start()
+		// Stop spinner after running function
+		defer func() { s.Stop() }()
+	}
+
+	err = pingSatelliteServer(satelliteUrl, s)
+	if err != nil {
+		return fmt.Errorf("unable to verify that given server is Satellite server: %v", err)
+	} else {
+		configureSatelliteResult.IsServerSatellite = true
+	}
+
+	// Create file for script file
+	satelliteScriptPath := path.Join(varLibDir, "katello-rhsm-consumer")
+	defer os.Remove(satelliteScriptPath)
+	scriptFile, err := os.Create(satelliteScriptPath)
+	if err != nil {
+		return fmt.Errorf("could not create %v file: %w", satelliteScriptPath, err)
+	}
+	defer scriptFile.Close()
+	err = os.Chmod(satelliteScriptPath, 0700)
+	if err != nil {
+		return fmt.Errorf("could not set permissions on %v file: %w", satelliteScriptPath, err)
+	}
+
+	err = downloadSatelliteScript(scriptFile, satelliteUrl, s)
+	if err != nil {
+		return fmt.Errorf("could not download satellite bootstrap script: %w", err)
+	}
+
+	if s != nil {
+		s.Suffix = fmt.Sprintf(" Configuring '%v' to use Satellite server: %v", hostname, satelliteUrl.Host)
+	}
+	// Run the script. It should install CA certificate, change configuration of rhsm.conf.
+	// In theory, it can do almost anything.
+	cmd := exec.Command(satelliteScriptPath)
+	err = cmd.Run()
+	if err != nil {
+		return fmt.Errorf("execution of %v script failed: %w", satelliteScriptPath, err)
+	} else {
```

Comment:
This else clause can be dropped and leave the `configureSatelliteResult.HostConfigured = true` line in the main function body.